src/ordered_dict.jl

```diff
@@ -18,8 +18,11 @@ mutable struct OrderedDict{K,V} <: AbstractDict{K,V}
     maxprobe::Int
     dirty::Bool
 
+    function OrderedDict{K,V}(slots, keys, vals, ndel, maxprobe, dirty) where {K,V}
+        new{K,V}(slots, keys, vals, ndel, maxprobe, dirty)
+    end
     function OrderedDict{K,V}() where {K,V}
-        new{K,V}(zeros(Int32,16), Vector{K}(), Vector{V}(), 0, 0, false)
+        OrderedDict{K,V}(zeros(Int32,16), Vector{K}(), Vector{V}(), 0, 0, false)
     end
     function OrderedDict{K,V}(kv) where {K,V}
```

Comment:
Can we now move this to be an outer constructor?
Possibly even we could move both to be outer constructors and just leave the default constructor behind (the one that just takes 1 arg per field)

Reply:
You're right, outer constructors are more elegant here. This has been changed